exporter/datadogexporter/internal/metrics/consumer.go

```diff
@@ -35,18 +36,22 @@ var _ translator.APMStatsConsumer = (*Consumer)(nil)
 // Consumer implements translator.Consumer. It records consumed metrics, sketches and
 // APM stats payloads. It provides them to the caller using the All method.
 type Consumer struct {
-	ms        []datadogV2.MetricSeries
-	sl        sketches.SketchSeriesList
-	as        []pb.ClientStatsPayload
-	seenHosts map[string]struct{}
-	seenTags  map[string]struct{}
+	ms                          []datadogV2.MetricSeries
+	sl                          sketches.SketchSeriesList
+	as                          []pb.ClientStatsPayload
+	seenHosts                   map[string]struct{}
+	seenTags                    map[string]struct{}
+	sumAsRate                   bool
+	sumToRateConversionInterval time.Duration
 }
 
-// NewConsumer creates a new Datadog consumer. It implements translator.Consumer.
-func NewConsumer() *Consumer {
+// NewConsumer creates a new Datadog consumer.
+func NewConsumer(metricsSumAsRate bool, metricsSumToRateConversionInterval time.Duration) *Consumer {
```

Comment:
For me, boolean args are a big no-go. They are ambiguous. When reading the calls it is impossible to understand what they are supposed to mean. Instead, maybe you'll want to use a structure (perhaps called `ConsumerOptions`).

However, I'd recommend to wait for @mx-psi's approval and review before committing any more work to this.